tools/oam-converter/pkg/app/confdata.go

```
@@ -5,201 +5,280 @@ package app
-		fmt.Println("Failed to read YAML file:", err)
-		return err
-	}
-
-	//Read Comp file
-	compData, err := ioutil.ReadFile("../../examples/bobs-books/bobs-books-comp.yaml")
-	if err != nil {
-		fmt.Println("Failed to read YAML file:", err)
-		return err
-	}
-	//A map for app file
-	appMap := make(map[string]interface{})
+	var inputDirectory string
+	var outputDirectory string
 
-	// Unmarshal the OAM YAML input data into the map
-	err = yaml.Unmarshal(appData, &appMap)
-	if err != nil {
-		log.Fatalf("Failed to unmarshal YAML: %v", err)
+	//Check the length of args
+	if len(os.Args) != 3 {
+		fmt.Println("Not enough args to run tool")
+		return nil
 	}
+	inputDirectory = os.Args[1]
+	outputDirectory = os.Args[2]
 
-	workloadTraitMap, componentNames, ingressTraits, metricsTraits := extract.HandleYAMLStructurePanic(appMap)
-	fmt.Print(ingressTraits)
-
-	//Splitting up the comp file with "---" delimiter into multiple objects
-	compStr := string(compData)
-	compObjects := strings.Split(compStr, "---")
+	//used to store app file data
+	var appData []map[string]interface{}
 
-	//Array of components in comp file
+	//used to store comp file data
 	var components []map[string]interface{}
 
-	for _, obj := range compObjects {
-		var component map[string]interface{}
-		err := yaml.Unmarshal([]byte(obj), &component)
-		if err != nil {
-			log.Fatalf("Failed to unmarshal YAML: %v", err)
-		}
-		components = append(components, component)
-	}
-	weblogicMap := make(map[string]*vzapi.VerrazzanoWebLogicWorkload)
-	coherenceWorkloads, helidonWorkloads, weblogicMap, err = segregateWorkloads(weblogicMap, components, componentNames, helidonWorkloads, coherenceWorkloads)
+	//used to store non-oam file data
+	var K8sResources []map[string]interface{}
+
+	//iterate through user inputted directory
+	files, err := iterateDirectory(inputDirectory)
 	if err != nil {
-		fmt.Printf("Failed to segregate: %v\n", err)
-		return err
+		fmt.Println("Error in iterating over directory", err)
```

should return an error here